Update build-intelligence.md (#8754)
harness/developer-hub: docs/continuous-integration/use-ci/build-and-upload-artifacts/build-intelligence.md

Build Intelligence is part of [Harness CI Intelligence](/docs/continuous-integration/get-started/harness-ci-intelligence), a suite of features in Harness CI designed to improve build times. It saves time by reusing outputs from previous builds. BI works by storing these outputs locally or remotely and retrieving them when inputs haven't changed. This process avoids the need to regenerate outputs, significantly speeding up the build process and enhancing efficiency.

:::note

> * Build Intelligence is currently only offered to Harness Cloud, self hosted Kubernetes Infra with support for other self hosted Infras coming soon. This feature is behind the feature flag `CI_CACHE_ENABLED` if infra is Harness Cloud. 

Reviewer: ```suggestion
* Build Intelligence is currently only offered to Harness Cloud, self hosted Kubernetes Infra with support for other self hosted Infras coming soon. This feature is behind the feature flag `CI_CACHE_ENABLED` when the infrastructure is Harness Cloud. 
```

2. **Cache Operations**: At the start of the build, the plugin registers with Gradle to check for cached build outputs. If available, it retrieves and provides them to Gradle, avoiding the need to regenerate them.

The above operation is transparent to you as a user and happens in the background. 

> (here we have to paste the SS of UI to enable buildIntelligence) 

Reviewer: ```suggestion
**Harness Cloud**:
When you enable Build Intelligence in the UI, it automatically injects a background step responsible for bringing up the proxy server, autodetecting the build tool, and injecting the configuration. If the Run or Tests steps are executed in a container, we will also automatically re-inject the configuration there.

**Kubernetes**:
When you enable Build Intelligence in the UI, it automatically injects a background step responsible for bringing up the proxy server. For every Run or Test step, we will autodetect the build tool and auto-inject the appropriate configuration based on it.
```

> Harness Cloud:

Reviewer: ```suggestion
```

>     If you enable buildIntelligence on UI, It will automatically inject the back ground step which is responsible to bring up the proxy-server and autodetect the build tool and inject the configuration. If the run/Tests steps runs in the container there also we will again auto-inject the config.

Reviewer: ```suggestion
```

> Kubernetes:

Reviewer: ```suggestion
```

>     If you enable buildIntelligence on UI, It will automatically inject the back ground step which is responsible to bring up the proxy-server.From every run/test steps we will auto detect and auto-inject the config based on the build tool.

Reviewer: ```suggestion
```

Currently the proxy server is running on port: `8082`

#### Configuration for Gradle

> 1. Import the build cache plugin in `settings.gradle` file: Or If you enable BuildInteligence on UI, the below configuration automatically injects in the below directories.

Reviewer: ```suggestion
1. Import the build cache plugin in the `settings.gradle` file. Alternatively, if you enable Build Intelligence in the UI, the configuration below is automatically injected into the following directories:

- $GRADLE_HOME/init.d/init.gradle
- $GRADLE_USER_HOME/init.d/init.gradle
- $HOME/.gradle/init.d/init.gradle (if `GRADLE_HOME` and `GRADLE_USER_HOME` are set in your environment variables).
```

> `$GRADLE_HOME/init.d/init.gradle, $GRADLE_USER_HOME/init.d/init.gradle, $HOME/.gradle/init.d/init.gradle`(If $GRADLE_HOME and $GRADLE_USER_HOME are present in your environments).

Reviewer: ```suggestion
```

> 2. If you want to use your own specific maven repository, you should give `MAVEN_URL` as stage variable or else It will pull from mavenCentral() where our HarnessGradlePlugin resides(for reference: https://central.sonatype.com/artifact/io.harness/gradle-cache/overview).

Reviewer: ```suggestion
2. If you want to use a custom Maven repository, you should set `MAVEN_URL` as a stage variable. Otherwise, it will pull from mavenCentral(), where the HarnessGradlePlugin is hosted. (For reference: https://central.sonatype.com/artifact/io.harness/gradle-cache/overview).
```

> 3. If you want to use your own maven repository, you should download the latest version plugin dependencies from maven central(As of now the latest version is `0.0.4`).

Reviewer: ```suggestion
3. To use your own Maven repository, download the latest version of the plugin dependencies from Maven Central. (As of now, the latest version is 0.0.4).
```

> 4. The variables `HARNESS_ACCOUNT_ID` and `HARNESS_CACHE_SERVICE_ENDPOINT` will be automatically set if you enable build Intelligence.

Reviewer: ```suggestion
4. The variables `HARNESS_ACCOUNT_ID` and `HARNESS_CACHE_SERVICE_ENDPOINT` will be automatically set when you enable Build Intelligence.
```
